src/backend/stream.rs

```diff
@@ -12,8 +12,15 @@ use pulse_ffi::*;
 use std::{mem, ptr};
 use std::ffi::{CStr, CString};
 use std::os::raw::{c_long, c_void};
+use std::slice;
+use ringbuf::RingBuffer;
 
 const PULSE_NO_GAIN: f32 = -1.0;
+// When running duplex callbacks, the input data is fed to a ring buffer, and then later copied to
+// a linear piece of memory is used to hold the input sample, so that they are passed to the audio
+// callback that delivers it to the users. Their size depends on the buffer size requested initially.
+// This is to be tuned when changing tlength and fragsize, but this value works for now.
```

Comment:
Add an assert or a comment where tlength/fragsize are set to connect these together.